### PR TITLE
Disable options to vote for an item belonging to the logged user

### PR DESCRIPTION
**Pull request by iuresti**
Not to show buttons to vote when the item belongs to the logged user
Add validation in back end to avoid logged user to vote for an item belonging to himself.

### CODE REVIEW

File: src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java [hunk 1]
```diff
@@ -74,31 +74,43 @@ public Answer get(final Long id) {
...
     @Override
     public void upVote(Long answerId, User currentUser) {
         Answer answer = get(answerId);
-        if (answer != null && ruleService.isValidUserPermission(RuleName.VOTED_UP_ANSWER, currentUser)) {
-            answer.setVotesUp(answer.getVotesUp() + 1);
-            answerRepository.save(answer);
+        User answerOwner = answer.getUser();
 
-            savePointsForAnswer(answer, RuleName.VOTED_UP_ANSWER);
+        if (!answerOwner.getId().equals(currentUser.getId())) {
```
Comment (wdonet) [inline src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java]: why not avoiding multiple ifs with something like:
if (answerOwner.getId().equals(currentUser.getId())) {
   log()
   return;
}
if (answer != null && ruleService.isValidUserPermission(RuleName.VOTED_UP_ANSWER, currentUser)) {
..
}
else { ... }

File: src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java [hunk 2]
```diff
@@ -78,27 +78,40 @@ public void save(Question question) {
     @Override
     public void downVote(Long questionId, User currentUser) {
         Question question = get(questionId);
-        if (question != null && ruleService.isValidUserPermission(RuleName.VOTED_DOWN_QUESTION, currentUser)) {
-            question.setVotesDown(question.getVotesDown() + 1);
-            questionRepository.save(question);
+        User questionOwner = question.getUser();
 
-            savePointsForQuestion(question, RuleName.VOTED_DOWN_QUESTION);
+        if (!questionOwner.getId().equals(currentUser.getId())) {
```
Comment (wdonet) [inline src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java]: same , look comment above

File: src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java [hunk 3]
```diff
@@ -78,27 +78,40 @@ public void save(Question question) {
...
     }
 
 
     @Override
     public void upVote(Long questionId, User currentUser) {
         Question question = get(questionId);
-        if (question != null && ruleService.isValidUserPermission(RuleName.VOTED_UP_QUESTION, currentUser)) {
-            question.setVotesUp(question.getVotesUp() + 1);
-            questionRepository.save(question);
+        User questionOwner = question.getUser();
```
Comment (wdonet) [inline src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java]: idem, look comment above

File: src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java (hunk 1 above)
Comment (iuresti) [inline src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java]: Fixed

File: src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java (hunk 2 above)
Comment (iuresti) [inline src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java]: fixed

File: src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java (hunk 3 above)
Comment (iuresti) [inline src/main/java/com/nearsoft/questions/service/impl/QuestionServiceImpl.java]: fixed @wdonet

File: src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java (hunk 1 above)
Comment (wdonet) [inline src/main/java/com/nearsoft/questions/service/impl/AnswerServiceImpl.java]: nice, thx!

**Pull request merged**